src/Microsoft.Health.Fhir.Core/Features/Operations/BulkUpdate/BulkUpdateOrchestratorJob.cs

```diff
@@ -0,0 +1,223 @@
+    {
+        private readonly IQueueClient _queueClient;
+        private readonly Func<IScoped<ISearchService>> _searchService;
+        private readonly CoreFeatureConfiguration _configuration;
+        private readonly ILogger<BulkUpdateOrchestratorJob> _logger;
+        private const int CoordinatorMaxDegreeOfParallelization = 4;
+        private const int NumberOfParallelRecordRanges = 100;
+        private const string OperationCompleted = "Completed";
+        private readonly List<string> _excludedResourceTypes = new() { "SearchParameter", "StructureDefinition" };
+
+        public BulkUpdateOrchestratorJob(
+            IQueueClient queueClient,
+            Func<IScoped<ISearchService>> searchService,
+            IOptions<CoreFeatureConfiguration> configuration,
+            ILogger<BulkUpdateOrchestratorJob> logger)
+        {
+            EnsureArg.IsNotNull(queueClient, nameof(queueClient));
+            EnsureArg.IsNotNull(searchService, nameof(searchService));
+            EnsureArg.IsNotNull(configuration, nameof(configuration));
+
+            _queueClient = queueClient;
+            _searchService = searchService;
+            _logger = logger;
+            _configuration = configuration.Value;
+        }
+
+        public async Task<string> ExecuteAsync(JobInfo jobInfo, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(jobInfo, nameof(jobInfo));
+
+            BulkUpdateDefinition definition = jobInfo.DeserializeDefinition<BulkUpdateDefinition>();
+            var surrogateIdRangeSize = (int)definition.MaximumNumberOfResourcesPerQuery;
+
+            _logger.LogJobInformation(jobInfo, "Loading job by Group Id.");
+            var groupJobs = await _queueClient.GetJobByGroupIdAsync(QueueType.BulkUpdate, jobInfo.GroupId, true, cancellationToken);
+
+            // If system level Parallel bulk update then create sub jobs by resourceType-surrogateId ranges
+            using var searchService = _searchService.Invoke();
+            if (definition.IsParallel && string.IsNullOrEmpty(definition.Type))
+            {
+                var atLeastOneWorkerJobRegistered = false;
+
+                // For system level bulk update, get all the resource types that are used in the system and enqueue jobs for each resource type-surrogateId range combination.
+                // For ResourceType based bulk updates query with includes and revinclude could return multiple resourceTypes so we will do the generic search without surrogateId ranges
+                var resourceTypes = await searchService.Value.GetUsedResourceTypes(cancellationToken);
+                resourceTypes = resourceTypes
+                    .Where(x => !_excludedResourceTypes.Contains(x))
+                    .ToList();
+
+                resourceTypes = resourceTypes.OrderByDescending(x => string.Equals(x, "Observation", StringComparison.OrdinalIgnoreCase)).ToList(); // true first, so observation is processed as soon as
```

Comment:
The Observation first thing was a leftover from export where we had noticed that observation was the most common resource type in general. This meant it would start getting processed first as it would take the longest. I don't know if this is a pattern worth continuing, but it also doesn't hurt anything.